sim/src/mechanics/intersection.rs

```diff
@@ -134,6 +135,7 @@ impl IntersectionSimState {
         }
         if self.break_turn_conflict_cycles {
             if let AgentID::Car(car) = agent {
+                //self.blocked_by.drain_filter(|(_, c)| *c == car);
```

Comment:
Please remove the tmp debugging bit

Reply:
actually, that drain_filter is a Rust experimental feature, which replaces this:
 retain_btreeset(&mut self.blocked_by, |(_, c)| *c != car); 
I was playing with it and forgot to remove the commented out line. Eventually, it should be used instead, although the predicate has opposite meaning, so caution should be used.